india_compliance/gst_india/overrides/party.py

```diff
@@ -164,3 +166,28 @@ def make_address(doc):
             "is_shipping_address": doc.get("is_shipping_address"),
         }
     ).insert()
+
+
+@frappe.whitelist()
+def validate_pancard_status(pan):
+    url = "https://eportal.incometax.gov.in/iec/servicesapi/getEntity"
+    payload = {
+        "aadhaarNumber": "887598175965",  # this is random generated aadhaar_number
+        "pan": pan,
+        "preLoginFlag": "Y",
+        "serviceName": "linkAadhaarPreLoginService",
+    }
+    response = requests.post(url, json=payload)
+    messages = response.json().get("messages")
+    if not messages:
+        return ""
+
+    error_code_desc_map = {
+        "EF40124": "Linked",  # pan linked to given aadhar card
```

Comment:
- [ ] PAN Doctype: Created only if a status is available
- [ ] Random Aadhar (Check performance)
- [ ] Aadhar Invalid Error
- [ ] Try Except: throw error with force refresh
- [ ] PAN Info in utils
- [ ] Status from Linked to Valid
- [ ] If status is available: No refresh button on the face of doc. However user can refresh after going to PAN doc.